scripts/factorio-build.ts

```diff
@@ -159,6 +159,14 @@ async function processMod(): Promise<void> {
   fs.mkdirSync(tempPath);
   fs.mkdirSync(tempIconsPath);
 
+  try {
```

Comment:
**question:** Why was this code added? I'm not sure I want to merge this into main.

Reply:
When following instructions at https://github.com/factoriolab/factoriolab/wiki/Creating-a-new-factorio-mod-set, the `factorio-build` step would fail because it would try to create a `data/2.0q/data.json` file while the `data/2.0q` directory did not exist yet. This code creates that directory as needed.

Reply:
Here is an alternative idea that also fixes another hurdle with those instructions. If you like it I can make another PR:

Unlike the rest of `data.json`, its `defaults` key is not auto-generated but is manually maintained. When following wiki instructions, it is initially missing which causes weird issues like assuming beacons don’t exist.

Its source of truth could be moved into new `data/*/defaults.json` files that can use indentation. `factorio-build.ts` would still copy its contents into `data.json` so that the web UI does not need to make an extra HTTP request.

Wiki instructions would gain a new step like:
> 4. Create a `src\data\abc` directory, replacing `abc` with your new mod id, then copy `src\data\2.0\defaults.json` to `src\data\abc\defaults.json` and tweak it as needed.

This chunk of code creating the directory would become irrelevant. An extra manual step is fine since defaults need manual intervention anyway.